snack/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+	"html/template"
+	"math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+func main() {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		snacks := [...]string{
+			"Spam Musubi",
+			"Pocky Sticks",
+			"Kasugai Gummy",
+			"Green Tea Mochi",
+			"Shrimp-flavored Chips",
+			"Red Bean Rice Cake",
+			"Pretz Sticks",
+			"Peaches in Agar Jelly",
+		}
+
+		rand.Seed(time.Now().Unix())
+		s := snacks[rand.Intn(len(snacks))]
+
+		t, err := template.ParseFiles(templatePath("index.tpl"))
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "error parsing template: %v\n", err)
+			return
+		}
+
+		err = t.Execute(w, s)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "error executing template: %v\n", err)
+			return
+		}
+	})
+
+	http.ListenAndServe(":8083", nil)
+}
+
+func templatePath(f string) string {
+	dir := os.Getenv("TEMPLATE_DIR")
+	if dir == "" {
+		dir = "snack/web/templates"
```

Comment:
I copied this snippet from the fortune service, but if the TEMPLATE_DIR is not set in the YAML this fallback dir doesn't actually work. I tried some different values like dir = "./web/templates" but those didn't work, either. Maybe there's something about what filepath considers as its root that I'm not understanding?

Reply:
yes! it depends on where you run the binary from. i think it's probably fine to panic if we don't have a TEMPLATE_DIR, since we never expect to run this outside of k8s

Reply:
you could also just hard-code the TEMPLATE_DIR here as /go/src/.../templates. 

one of the reasons i'm a big believer in containers is that if you only run your server inside a container, then you don't have to worry about these fiddly configuration options